test/ts/ssh-test/reconnectTests.ts

```diff
@@ -747,4 +755,111 @@ export class ReconnectTests {
 			);
 		}
 	}
+
+	@test
+	public async reconnectWhileStreaming() {
+		const [clientStream, serverStream] = await connectSessionPair(
+			this.clientSession,
+			this.serverSession,
+		);
+		await this.waitUntilReconnectEnabled();
+		const [serverChannel, clientChannel] = await this.initializeChannelPair();
+
+		// Start continuously sending/receiving data with both client and server sessions.
+		const sendAndReceiveUntilEnd = async (
+			channel: SshChannel,
+			cancellation: CancellationToken,
+		): Promise<number> => {
+			const stream = new SshStream(channel);
+			cancellation.onCancellationRequested(() => stream.end());
+
+			let receiveCount = 0;
+			const sendBuffer = Buffer.alloc(4);
+			let receiveBuffer = Buffer.alloc(0);
+			stream.on('data', (data) => (receiveBuffer = Buffer.concat([receiveBuffer, data])));
+			const streamErrorPromise = new Promise<void>((_, reject) => stream.once('error', reject));
+
+			while (true) {
+				const writer = new SshDataWriter(sendBuffer);
+				writer.writeUInt32(receiveCount);
+				const readPromise = new Promise<void>((resolve, reject) => {
+					if (receiveBuffer.length === 0) {
+						stream.once('data', resolve);
+					} else {
+						resolve();
+					}
+				});
+
+				let sent = false;
+				try {
+					await Promise.race([
+						new Promise<void>((resolve, reject) => {
+							if (!stream.write(sendBuffer, (e) => (e ? reject(e) : resolve()))) {
+								stream.once('drain', resolve);
+							}
+						}),
+						streamErrorPromise,
+					]);
+					sent = true;
+
+					await Promise.race([readPromise, streamErrorPromise]);
+					const reader = new SshDataReader(receiveBuffer);
+					assert.strictEqual(reader.readUInt32(), receiveCount);
```

Comment:
Do we want to do anything to verify how many times this is called? I'm just thinking about testing scenarios where the assert is skipped because something above it changed the execution flow. I don't want to make this too complicated - just checking if it's important.

Reply:
The exact number of times will vary depending on timing. There are assertions below (lines 841-842, 854-855) that verify at least _some_ messages were sent/received both before and after reconnection.